Tests/AsyncHTTPClientTests/HTTPClientTests.swift

```diff
@@ -3525,4 +3525,47 @@ final class HTTPClientTests: XCTestCaseHTTPClientTestsBaseClass {
         let response = try client.get(url: self.defaultHTTPBinURLPrefix + "get").wait()
         XCTAssertEqual(.ok, response.status)
     }
+
+    func testAsyncExecuteWithCustomTLS() async throws {
+        let httpsBin = HTTPBin(.http1_1(ssl: true))
+        defer {
+            XCTAssertNoThrow(try httpsBin.shutdown())
+        }
+
+        // A client with default TLS settings, i.e. it won't accept `httpsBin`'s fake self-signed cert
+        let client = HTTPClient(eventLoopGroup: MultiThreadedEventLoopGroup.singleton)
+        defer {
+            XCTAssertNoThrow(try client.shutdown().wait())
+        }
+
+        var request = HTTPClientRequest(url: "https://localhost:\(httpsBin.port)/get")
+
+        // For now, let's allow bad TLS certs
+        request.tlsConfiguration = TLSConfiguration.clientDefault
+        request.tlsConfiguration!.certificateVerification = .none
+
+        let response1 = try await client.execute(request, timeout: .hours(99))
+        XCTAssertEqual(.ok, response1.status)
+
+        // For the second request, we reset the TLS config
+        request.tlsConfiguration = nil
+        do {
+            let response2 = try await client.execute(request, timeout: .hours(99))
+            XCTFail("shouldn't succeed, self-signed cert: \(response2)")
+        } catch {
+            switch error as? NIOSSLError {
+            case .some(.handshakeFailed(_)):
+                () // ok
+            default:
+                XCTFail("unexpected error: \(error)")
+            }
+        }
+
+        // And finally we allow it again.
+        request.tlsConfiguration = TLSConfiguration.clientDefault
+        request.tlsConfiguration!.certificateVerification = .none
```

Comment:
lets get rid of the force unwrap.
```suggestion
        // And finally we allow it again.
        let tlsConfiguration = TLSConfiguration.clientDefault
        tlsConfiguration.certificateVerification = .none
        request.tlsConfiguration = tlsConfiguration
```

Reply:
I mean I can do that but what's bad about force unwraps used correctly? This one won't trigger

Reply:
I'm assigning it literally the line above. If this fails then AHC has a severe bug and I'd like to know about it rather than hide it.

Reply:
I'm not against using force unwraps in general. However, you should only use them if you can reason about it locally. `tlsConfigurataion` isn't a local variable but a property of a type that is define in another module. Keep in mind that properties in Swift can be computed so it can be arbitrary code that is executed on the getter and setter. If you want to know that the force unwrap is safe, you and everyone that looks at that tests in the future, needs to look at the implementation.

In theory you can also add a comment to the force unwrap explaining why it is safe but the code that circumvents it is straight forward and also just a single line.

Reply:
@dnadoba Apologies but this is backwards. 

If a user writes

```
        request.tlsConfiguration = TLSConfiguration.clientDefault
        request.tlsConfiguration!.certificateVerification = .none
```

then this has to work. Yes, it is _possible_ to break this code if you were so inclined but then the `HTTPClientRequest` type would just be outright broken.

Remember, it's not the goal to be user hostile. If this type were defined in another package that was known as having user hostile types then your comment might be appropriate. But this is a type from this package and it shouldn't be user hostile.

Reply:
added the comment. The guidelines aren't meant for test code but for production code.